client/src/components/chess-game/ChessGame.js

```diff
@@ -1,11 +1,77 @@
 import React from 'react';
+import Chess from 'chess.js';
 import ChessBoard from '../chess-board/ChessBoard';
+import { WHITE_PLAYER } from '../../config/chess-game';
+import './ChessGame.css';
 
-const ChessGame = () => (
-  <div>
-    ChessGame Component
-    <ChessBoard />
-  </div>
-);
+const INIT_STATE = {
+  board: [],
+  currentPlayer: WHITE_PLAYER,
+  isGameOver: false,
+  gameOverMessage: '',
+};
 
-export default ChessGame;
+export default class ChessGame extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = Object.assign({}, INIT_STATE);
+    this.startGame = this.startGame.bind(this);
+    this.movePiece = this.movePiece.bind(this);
+    this.updateBoardState = this.updateBoardState.bind(this);
+    this.calcPossibleMoves = this.calcPossibleMoves.bind(this);
+  }
+
+  componentDidMount() {
+    this.startGame();
+  }
+
+  startGame() {
+    this.chess = new Chess();
+    this.squares = this.chess.SQUARES;
+    const board = this.chess.board();
+    const currentPlayer = this.chess.turn();
+    this.setState(Object.assign({}, INIT_STATE, { board, currentPlayer }));
+  }
+
+  movePiece(from, to) {
+    this.chess.move({ from, to });
+    this.updateBoardState();
+  }
+
+  updateBoardState() {
+    const board = this.chess.board();
+    const currentPlayer = this.chess.turn();
+    this.setState({ board, currentPlayer });
+  }
+
+  calcPossibleMoves(param) {
+    return this.chess.moves(param);
+  }
+
+  render() {
+    const {
+      board,
+      isGameOver,
+      gameOverMessage,
+      currentPlayer,
+    } = this.state;
+    const { squares } = this;
+    return (
+      <>
+        <div id="chess-game-container">
+          <ChessBoard
+            calcPossibleMoves={this.calcPossibleMoves}
+            movePiece={this.movePiece}
+            squares={squares}
+            board={board}
+            currentPlayer={currentPlayer}
```

Comment:
The currentPlayer prop is not required by chessBoard. Instead, you need to pass the playerColor prop (black or white) when the game is initialized.

Reply:
In code base in ChessBoard component I am not able to what is the use of playerColor prop is it required, or does it means color of current player ?

Reply:
Yes, It will be used to render the board based on the piece color of the player. 
In the case of white, the upper right corner of the board is a8 and in case of black, it is a1.

Reply:
Ok, I will suggest we should have game config constants, In this PR [chess-game.js](https://github.com/pesto-students/batch-6-chess-multiplayer/blob/b232a4df2418b2864dbb691e501da33c75756bb4/client/src/config/chess-game.js) file we can use that as constants for entire app, instead of writing different for different components.